split_to_CMYK.jsx

```diff
@@ -1 +1,124 @@
+    };
+    var s2t = function (s) {
+        return app.stringIDToTypeID(s);
+    };
+    var descriptor = new ActionDescriptor();
+    var descriptor2 = new ActionDescriptor();
+    var descriptor3 = new ActionDescriptor();
+    var reference = new ActionReference();
+
+    reference.putClass( s2t( "adjustmentLayer" ));
+    descriptor.putReference( c2t( "null" ), reference );
+    descriptor2.putString( s2t( "name" ), (layerName) + """ Levels""" );
+    descriptor2.putBoolean( s2t( "group" ), true );
+    descriptor3.putEnumerated( s2t( "presetKind" ), s2t( "presetKindType" ), s2t( "presetKindDefault" ));
+    descriptor2.putObject( s2t( "type" ), s2t( "levels" ), descriptor3 );
+    descriptor.putObject( s2t( "using" ), s2t( "adjustmentLayer" ), descriptor2 );
+    executeAction( s2t( "make" ), descriptor, DialogModes.NO );
+//add hue sat adj layer - if statement is so Black doesnt get a hue adj layer
+    if (layerName != 'Black') {
+    var c2t = function (s) {
+        return app.charIDToTypeID(s);
+    };
+    var s2t = function (s) {
+        return app.stringIDToTypeID(s);
+    };
+    var descriptor = new ActionDescriptor();
+    var descriptor2 = new ActionDescriptor();
+    var descriptor3 = new ActionDescriptor();
+    var reference = new ActionReference();
+
+    reference.putClass( s2t( "adjustmentLayer" ));
+    descriptor3.putReference( c2t( "null" ), reference );
+    descriptor.putString( s2t( "name" ), (layerName) + """ color adjust""" );
+    descriptor.putBoolean( s2t( "group" ), true );
+    descriptor2.putEnumerated( s2t( "presetKind" ), s2t( "presetKindType" ), s2t( "presetKindDefault" ));
+    descriptor2.putBoolean( s2t( "colorize" ), true );
+    descriptor.putObject( s2t( "type" ), s2t( "hueSaturation" ), descriptor2 );
+    descriptor3.putObject( s2t( "using" ), s2t( "adjustmentLayer" ), descriptor );
+    executeAction( s2t( "make" ), descriptor3, DialogModes.NO );
+//sets hue, sat, lightness
+    var descriptor = new ActionDescriptor();
+    var descriptor2 = new ActionDescriptor();
+    var descriptor3 = new ActionDescriptor();
+    var list = new ActionList();
+    var reference = new ActionReference();
+
+    reference.putEnumerated( s2t( "adjustmentLayer" ), s2t( "ordinal" ), s2t( "targetEnum" ));
+    descriptor.putReference( c2t( "null" ), reference );
+    descriptor2.putEnumerated( s2t( "presetKind" ), s2t( "presetKindType" ), s2t( "presetKindCustom" ));
+    descriptor3.putEnumerated( s2t( "channel" ), s2t( "channel" ), s2t( "composite" ));
+    descriptor3.putInteger( s2t( "hue" ), hue );
+    descriptor3.putInteger( c2t( "Strt" ), sat );
+    descriptor3.putInteger( s2t( "lightness" ), light );
+    list.putObject( s2t( "hueSatAdjustmentV2" ), descriptor3 );
+    descriptor2.putList( s2t( "adjustment" ), list );
+    descriptor.putObject( s2t( "to" ), s2t( "hueSaturation" ), descriptor2 );
+    executeAction( s2t( "set" ), descriptor, DialogModes.NO );
+    } else {return}
+}
+
+makeLayers();
```

Comment:
One of the unwritten best practices for PS and AI scripting is to keep all your executions within a function instead of in the global space.  Garbage collection in the CC apps is pretty awful, so it's easy to pollute the global space with variables that get stuck in memory for the session.  As an example, this is a real thing that can happen in an Adobe app:

- Create & run a script that defines a naked variable: `var number = 5;`
- 5 hours later (assuming you haven't closed Illustrator in the meantime), create & run a script that ONLY has this line: `alert(number)`

On running that second script, you'll see an alert window with the number 5 shown, even though you didn't even define a variable in that second script.  It makes debugging a huuuuuuuuuuuge nightmare.  Here's how I'd probably format lines 109-122, taking into account some of the suggestions I made above:
```suggestion
function runScript() {
  var doc = app.activeDocument;
  doc.changeMode(ChangeMode.CMYK);
  makeLayers();
  splitChannels('Cyan');
  splitChannels('Magenta');
  splitChannels('Yellow');
  splitChannels('Black');
  doc.ChangeMode(ChangeMode.RGB);
  addLevelsHueSat('Cyan', 174, 50, 20);
  addLevelsHueSat('Magenta', 300, 50, 20);
  addLevelsHueSat('Yellow', 50, 50, 20);
  addLevelsHueSat('Black')
  alert([...alert message...]);
}

runScript();
```

This will help make sure your vars & functions stay explicitly defined in the local scope, making sure they get destroyed after the script execution is finished.